### PR TITLE
Version 0.1.0-alpha

### PR DESCRIPTION
**Pull request by lfdebrux**
## New features

- Add tests for all components
- Add an extension `NunjucksExtension` that preprocesses Nunjucks templates
- Add an subclass of Undefined, `NunjucksUndefined`, that deals with instances where Nunjucks is more forgiving that Jinja
- Create an example of a Jinja Environment that handles almost all incompatibilities between Nunjucks and Jinja in the govuk-frontend templates
- Add a command-line tool, `scripts/njk-to-j2.py` that can be used for debugging the extension

## Bugfixes

- Fix bug with test helper `normalise_whitespace`

### CODE REVIEW

**Comment by risicle** [comment]
I can't tell from a brief skim, but is this industrial-strength enough to be split out into a totally govuk-unrelated `jinja-nunjucks` package?

File: tests/components/test_character_count.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+<div class="govuk-form-group govuk-form-group--error">
+  <label class="govuk-label" for="exceeding-characters">
+    Full address
+  </label>
+
+  <span id="exceeding-characters-error" class="govuk-error-message">
+    Please do not exceed the maximum allowed limit
+  </span>
+
+  <textarea class="govuk-textarea govuk-textarea--error js-character-count  govuk-textarea--error" id="exceeding-characters" name="exceeding" rows="5" aria-describedby="exceeding-characters-error">221B Baker Street
```
Comment (risicle) [inline tests/components/test_character_count.py]: I wonder whether you'll see quite a bit of churn over some of this output as the design system update some of their stuff - e.g. just from trivial stuff like class order...

Guess if it's trivial it'll be trivial to update the tests.

**Comment by risicle** [comment]
Neat! :+1:

File: tests/components/test_character_count.py (hunk 1 above)
Comment (lfdebrux) [reply tests/components/test_character_count.py]: The plan for the future is to make these test autogenerate from the govuk-frontend (that's what they do).

As a first pass copying and pasting was easier and quicker.

**Pull request merged**